shells/browser/shared/src/main.js

```diff
@@ -168,51 +162,39 @@ function createPanelIfReactLoaded() {
         container._hasInitialHTMLBeenCleared = true;
       }
 
-      function maybeSetReactSelectionFromBrowser() {
+      function setReactSelectionFromBrowser() {
         // When the user chooses a different node in the browser Elements tab,
         // copy it over to the hook object so that we can sync the selection.
         chrome.devtools.inspectedWindow.eval(
-          // Don't reset selection if it didn't change in Elements tab.
-          '(window.__REACT_DEVTOOLS_GLOBAL_HOOK__.$0 !== $0)' +
-            '  ? (window.__REACT_DEVTOOLS_GLOBAL_HOOK__.$0 = $0, true)' +
-            '  : false',
-          (didChangeSelection, error) => {
+          '(window.__REACT_DEVTOOLS_GLOBAL_HOOK__.$0 = $0, undefined)',
+          (_, error) => {
             if (error) {
               console.error(error);
-            } else if (didChangeSelection) {
+            } else {
               bridge.send('syncSelectionFromNativeElementsPanel');
             }
           }
         );
       }
 
-      let hasReactSelectionChanged = false;
-
-      function maybeSetBrowserSelectionFromReact() {
-        // Don't change the browser element selection when navigating away
-        // from the Components tab if the user didn't change the React selection.
-        if (!hasReactSelectionChanged) {
+      // When the user selects another item in the native Elements tab,
+      // select the corresponding React component.
+      let isListeningToNativeSelectionChange = false;
+      function ensureListeningToNativeSelectionChange() {
+        if (isListeningToNativeSelectionChange) {
           return;
         }
-        hasReactSelectionChanged = false;
-        chrome.devtools.inspectedWindow.eval(
-          '(window.__REACT_DEVTOOLS_GLOBAL_HOOK__.$0 != null)' +
-            '  ? (inspect(window.__REACT_DEVTOOLS_GLOBAL_HOOK__.$0), true)' +
-            '  : false',
-          (_, error) => {
-            if (error) {
-              console.error(error);
-            }
-          }
-        );
+        isListeningToNativeSelectionChange = true;
+        setReactSelectionFromBrowser();
+        chrome.devtools.panels.elements.onSelectionChanged.addListener(() => {
+          setReactSelectionFromBrowser();
```

Comment:
Hm... I understand why we had to remove the "two-way binding" but now we're updating the selection every time it changes in the Elements panel, rather than just when the React tab is about to be shown again? Why?

Reply:
Oh, I see, you want to avoid calling the bridge? I'll fix that.